pages/upgrade-guide/1.5.md

```diff
@@ -0,0 +1,306 @@
+```js
+$( "#popup" ).popup( {
+	classes: {
+		"ui-popup": "ui-corner-all awesome-fade-class",
+		"ui-popup-close-button": null
+	}
+} );
+```
+
+## Bolean style options
+
+jQuery Mobile has many include many boolean options on its widgets that simpliy toggled a class on the widget to change the display `corners` and `shadow` are great examples of this.
+
+because it is now with changes to our markup structure, very easy to do this either in your markup or now by using the classes option all of these options are now deprecated. to toggle these effects simply use the class option as demonstrated above
+
+## Theme
+
+### Icons
+
+In 1.4 we switched from using a `span` element for icons to using a `:before` or `:after` pseudo element so that icons could be added with just a class. We were very excited about this at the time but have since realized this causes many problems too. It is less performant, creates an awkward api, cant be target with JavaScript... So after careful consideration with the jQuery UI team, the Mobile team, a lot of reasearch and the creation of a whole new project [(The CSS Chassis Project)](https://github.com/jquery/chassis) so that we dont have to worry about this any more :-) We have decided it is best to switch back to using spans for icons. We understand the pain this causes in the upgrade process and apologize, but we felt this was this best approach moving forward for jQuery Mobile.
+
+In addition to the change back to spans we have re thought the way icons are positioned in an effort to both simplify our markup and css and at the same time make it more flexible icons are also now no longer absolutely positioned. Icons are now placed inline in the text content of where ever you would like them to be. This means you can now add as many icons to an element as you want and place them any where you would like. This makes icons much more flexible in your designs.
+
+To help you with some common positions that are a little more complicated then just inline we have also added a few icon helper classes `ui-widget-icon-block` will make you icon centered and display block. This is useful for top an bottom icons. Since icons are now inline elements by default they will right, center, or left justify with your text content. To help with positioning an icon all the way left or right independant of the text we have also added the classes `ui-widget-icon-floatbegining` and `ui-widget-icon-floatend`
+
+You can see examples of the new and old markup for icons in the [Button](#Button) section below
+
+## Enhancer
+
+In 1.4 auto enhancment was handled by a combination of a widget registry in the same file as the page widget, a custom method in the helpers module, and an extension to the base widget factory.
+
+We did a complete rethink of how auto enhancment with jQuery should work and the new module is FAST depending on the number of elements needing methods called its actually faster then directly calling each
```

Comment:
We did a complete rethink of how [~~auto enhancment~~autoenhancement or auto-enhancement or automatic enhancement] with jQuery should work and the new module is FAST[.] [~~d~~D]epending on the number of elements needing methods called its actually faster th[~~e~~a]n directly calling each